src/d3a/models/strategy/area_agents/two_sided_pay_as_bid_engine.py

```diff
@@ -102,21 +101,26 @@ def _match_offers_bids(self):
             original_bid_rate = bid.original_bid_price / bid.energy
             matched_rate = bid.price / bid.energy
 
+            trade_bid_info = [
+                original_bid_rate, bid.price/bid.energy,
+                offer.original_offer_price/offer.energy, offer.price/offer.energy,
+                original_bid_rate]
```

Comment:
Correct me if Im wrong, shouldn't the last value of _trade_bid_info_ list be _matched_rate_ rather than _original_bid_rate_ ?

Reply:
Not according to the documentation:

> determine trade_price to be original_bid (according PAB);

if market is clearing market (i.e. the match was made and cleared at this market / trade was created here): trade_price is original_bid. save this trade_price in newly created trade object. 

if market received a trade object from adjacent market: use trade_price provided in trade object